### PR TITLE
Add memory allocation tests

### PR DESCRIPTION
**Pull request by jhalakpatel**
Add memory allocation tests to ensure we do not implicitly copy input data.

This change is dependent on the corresponding MLIR-TensorRT PR: https://github.com/NVIDIA/TensorRT-Incubator/pull/73
Signed-off-by: Jhalak Patel [jhalakp@nvidia.com](mailto:jhalakp@nvidia.com)

### CODE REVIEW

File: tripy/tests/common/test_array.py [hunk 1]
```diff
@@ -176,3 +176,31 @@ def test_array_supported_python_sequence_type(self, dtype):
...
+
+        # (torch) Allocate on gpu -> gpu view
+        torch_data = torch.tensor(data).to(torch.device("cuda"))
+        a = Array(torch_data, None, tp.int32, tp.device("gpu"))
+        assert torch_data.data_ptr() == a.memref_value.ptr
+
+        # (cupy) Allocate on gpu -> gpu view
+        cp_data = cp.ones((1000,), dtype=np.int32)
+        a = Array(cp_data, None, tp.int32, tp.device("gpu"))
+        assert cp_data.data.ptr == a.memref_value.ptr
```
Comment (pranavm-nvidia) [inline tripy/tests/common/test_array.py:206]: Are we getting any different coverage with these? If we are, could we split them into separate tests?

Comment (yizhuoz004) [reply tripy/tests/common/test_array.py:206]: We have tests for the same purpose: `tripy/tests/integration/test_functional.py::test_tensor_round_tripping`. Only difference is that it's at the Tensor level.

If we are going to drop `Array`, then these tests probably are not needed.

Comment (jhalakpatel) [reply tripy/tests/common/test_array.py:206]: Agreed. We can either remove them completely or add a version similar to the one I added here: https://github.com/NVIDIA/TensorRT-Incubator/pull/73/files#diff-77cb958600670760fcd82a8ba97caf1a3ab9b9f14ff4f1b06513ea0a61e85019R243

Comment (parthchadha) [reply tripy/tests/common/test_array.py:206]: @jhalakpatel can we close this PR if the tests are no longer needed?

**Comment by jhalakpatel** [comment]
Closing this MR since we no longer have an Array class.

**Pull request closed without merging**